src/containers/StructurePage/components/EditFilters.jsx

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (c) 2017-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button } from 'ndla-ui';
+import { Pencil, Plus } from 'ndla-icons/action';
+import { DeleteForever } from 'ndla-icons/editor';
+import InlineEditField from './InlineEditField';
+import {
+  fetchSubjectFilters,
+  createSubjectFilter,
+  editSubjectFilter,
+  deleteFilter,
+} from '../../../modules/taxonomy';
+import WarningModal from '../../../components/WarningModal';
+
+class EditFilters extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      editMode: '',
+      filters: [],
+    };
+    this.addFilter = this.addFilter.bind(this);
+    this.getFilters = this.getFilters.bind(this);
+    this.showDeleteWarning = this.showDeleteWarning.bind(this);
+    this.deleteFilter = this.deleteFilter.bind(this);
+  }
+
+  componentDidMount() {
+    this.getFilters();
+  }
+
+  async getFilters() {
+    const filters = await fetchSubjectFilters(this.props.id);
+    this.setState({ filters, error: '' });
+  }
+
+  async addFilter(name) {
+    try {
+      await createSubjectFilter(this.props.id, name);
+      this.getFilters();
+      return true;
+    } catch (e) {
+      this.setState({ error: e.message });
+      return false;
+    }
+  }
+
+  async editFilter(id, name) {
+    try {
+      await editSubjectFilter(id, this.props.id, name);
+      this.getFilters();
+      return true;
+    } catch (e) {
+      this.setState({ error: e.message });
+      return false;
+    }
+  }
+
+  showDeleteWarning(filterId) {
+    this.setState({ showDelete: filterId });
+  }
+
+  async deleteFilter() {
+    try {
+      await deleteFilter(this.state.showDelete);
+      this.getFilters();
+    } catch (e) {
+      this.setState({ error: e });
+    }
+    this.setState({ showDelete: false });
+  }
+
+  render() {
+    const { classes, t } = this.props;
+    const { editMode, filters, showDelete } = this.state;
+
+    return (
```

Comment:
Noe logisk mulighet for å splitte opp denne komponenten eller gir ikke dette mening? Er flere sjekker nedover som kanskje ser litt mer ryddig ut hvis man splitter det opp i flere komponenter?